src/Microsoft.DotNet.Build.VstsBuildsApi/VstsBuildClient.cs

```diff
@@ -0,0 +1,251 @@
+            {
+               return (JObject)JToken.ReadFrom(jsonReader);
+            }
+        }
+
+        /// <summary>
+        /// Update VSTS Build Definition
+        /// </summary>
+        /// <returns>updated definitions id</returns>
+        private async Task<JObject> VstsUpdateDefinitionAsync(JObject definition)
+        {
+            string requestUri = $"DefaultCollection/{VstsBuildsApiBase(definition)}/_apis/build/definitions/{definition["id"].ToString()}?api-version=2.0";
+
+            HttpContent content = new StringContent(JsonConvert.SerializeObject(definition), System.Text.Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await GetClient(definition["project"]["url"].ToString()).PutAsync(requestUri, content);
+            ProcessResponseStatusCode(response);
+            return JObject.Parse(await response.Content.ReadAsStringAsync());
+        }
+
+        /// <summary>
+        /// Create VSTS Build Definition
+        /// </summary>
+        private async Task<JObject> VstsCreateDefinitionAsync(JObject definition)
+        {
+            /* Remove definition instance identifiable information */
+            RemoveIdentifiableInformation(definition);
+
+            string requestUri = $"DefaultCollection/{VstsBuildsApiBase(definition)}/_apis/build/definitions?api-version=2.0";
+
+            HttpContent content = new StringContent(JsonConvert.SerializeObject(definition), Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await GetClient(definition["project"]["url"].ToString()).PostAsync(requestUri, content);
+            ProcessResponseStatusCode(response);
+            return JObject.Parse(await response.Content.ReadAsStringAsync());
+
+        }
+
+        /// <summary>
+        /// Retrieve a VSTS Build Definition by name and VSTS folder path
+        /// </summary>
+        private async Task<JObject> VstsRetrieveDefinitionByIdAsync(JObject definition)
+        {
+            string requestUri = $"DefaultCollection/{VstsBuildsApiBase(definition)}/_apis/build/definitions/{definition["id"].ToString()}?api-version=2.0";
+
+            HttpResponseMessage response = await GetClient(definition["project"]["url"].ToString()).GetAsync(requestUri);
+            ProcessResponseStatusCode(response);
+            return JObject.Parse(await response.Content.ReadAsStringAsync());
+        }
+
+        /// <summary>
+        /// Retrieve a VSTS Build Definition by name and VSTS folder path
+        /// </summary>
+        private async Task<IReadOnlyList<JObject>> VstsRetrieveDefinitionsListByNameAndPathAsync(JObject definition)
+        {
+            string requestUri = $"DefaultCollection/{VstsBuildsApiBase(definition)}/_apis/build/definitions?api-version=2.0&name={definition["name"].ToString()}";
+
+            HttpResponseMessage response = await GetClient(definition["project"]["url"].ToString()).GetAsync(requestUri);
```

Comment:
nit: Might want to create a helper method for "project" "url" as it appears to be duplicated a few places, simila to VstsBuildsApiBase.